Link error in itk which installed by vcpkg

****

**Is your feature request related to a problem? Please describe.**
I use example code in https://itk.org/ITKExamples/src/Segmentation/Watersheds/SegmentWithWatershedImageFilter/Documentation.html?highlight=watershed
and report unresolved symbol error in linking.
I report it to ITK community and they say it seem to be a error in vcpkg.
https://discourse.itk.org/t/unresolved-symbol-when-using-watershed/4444/4
I push my cmakelists.txt in this linke which also contain details report of error.

**Proposed solution**
It should work when I using watershed algorithm. 
Now vcpkg tell me to use
`target_link_libraries(Reconstruction ITKFFT ITKVTK ITKznz itksys )`
to use itk. But watershed algorithm seem to be a part of other librariry.

**Describe alternatives you've considered**
In itk's document, using below code to include itk in cmakelists.txt


`find_package(ITK CONFIG REQUIRED)`
`include(${ITK_USE_FILE})`
`target_link_libraries(Reconstruction ${ITK_LIBRARIES})`
But it fail to link too. Error is below
`1> [CMake] CMake Error at D:/cmake/vcpkg/scripts/buildsystems/vcpkg.cmake:537 (_add_executable):
1> [CMake]   Target "Reconstruction" links to target "GTest::GTest" but the target was
1> [CMake]   not found.  Perhaps a find_package() call is missing for an IMPORTED
1> [CMake]   target, or an ALIAS target is missing?
1> [CMake] Call Stack (most recent call first):
1> [CMake]   CMakeLists.txt:43 (add_executable)
1> [CMake] CMake Error at D:/cmake/vcpkg/scripts/buildsystems/vcpkg.cmake:537 (_add_executable):
1> [CMake]   Target "Reconstruction" links to target "GTest::Main" but the target was
1> [CMake]   not found.  Perhaps a find_package() call is missing for an IMPORTED
1> [CMake]   target, or an ALIAS target is missing?
1> [CMake] Call Stack (most recent call first):
1> [CMake]   CMakeLists.txt:43 (add_executable)
1> [CMake] -- Generating done
1> [CMake] CMake Generate step failed.  Build files cannot be regenerated correctly.`

--- a/ports/itk/dont-build-gtest.patch
+++ b/ports/itk/dont-build-gtest.patch
@@ -1,0 +1,12 @@
+﻿diff --git a/Modules/ThirdParty/GoogleTest/itk-module.cmake b/Modules/ThirdParty/GoogleTest/itk-module.cmake
+index 5571748129..19ebb60885 100644
+--- a/Modules/ThirdParty/GoogleTest/itk-module.cmake
++++ b/Modules/ThirdParty/GoogleTest/itk-module.cmake
+@@ -3,6 +3,7 @@ Google's C++ test framework.")
+ 
+ itk_module(ITKGoogleTest
+   DEPENDS
++  EXCLUDE_FROM_DEFAULT
+   DESCRIPTION
+     "${DOCUMENTATION}"
+ )

--- a/ports/itk/portfile.cmake
+++ b/ports/itk/portfile.cmake
@@ -14,6 +14,7 @@ vcpkg_from_github(
         wrapping.patch
         opencl.patch
         use-the-lrintf-intrinsic.patch
+        dont-build-gtest.patch
 )
 
 vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS

--- a/ports/itk/vcpkg.json
+++ b/ports/itk/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "itk",
   "version": "5.2.1",
-  "port-version": 1,
+  "port-version": 2,
   "description": "Insight Segmentation and Registration Toolkit (ITK) is used for image processing and analysis.",
   "homepage": "https://github.com/InsightSoftwareConsortium/ITK",
   "license": "Apache-2.0",
@@ -14,7 +14,6 @@
       "default-features": false
     },
     "gdcm",
-    "gtest",
     {
       "name": "hdf5",
       "default-features": false,

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -3130,7 +3130,7 @@
     },
     "itk": {
       "baseline": "5.2.1",
-      "port-version": 1
+      "port-version": 2
     },
     "itpp": {
       "baseline": "4.3.1",

--- a/versions/i-/itk.json
+++ b/versions/i-/itk.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "e22b353e8f00bcdc12b78d414a712a3498a2d0d6",
+      "version": "5.2.1",
+      "port-version": 2
+    },
+    {
       "git-tree": "21beac1c375ce633f869566345d7418849633c3e",
       "version": "5.2.1",
       "port-version": 1

